src/sentry/api/endpoints/auth_index.py

```diff
@@ -131,6 +135,16 @@ def put(self, request):
                 status=status.HTTP_403_FORBIDDEN,
             )
 
+        if request.user.is_superuser and not is_active_superuser(request):
+            # if a superuser hitting this endpoint is not active, they are most likely
+            # trying to become active, and likely need to re-identify with SSO to do so.
+            redirect = request.META.get("HTTP_REFERER", "")
+            if not is_safe_url(redirect, allowed_hosts=(request.get_host(),)):
+                redirect = None
+
+            initiate_login(request, redirect)
+            raise SsoRequired(Organization.objects.get_from_cache(id=Superuser.org_id))
```

Comment:
everything lgtm, just to double check the flow, if the user is a superuser and not active, we set the redirect in the _next in sessions and they are redirected once SsoRequired is raised.

Reply:
yup, exactly.